src/Dialect/Krnl/KrnlOps.td

```diff
@@ -199,9 +199,38 @@ def KrnlGlobalOp : Op<Krnl_Dialect, "global"> {
     Operation for holding global data values.
   }];
 
-  let arguments = (ins AnyAttr:$shape, StrAttr:$name, OptionalAttr<AnyAttr>:$value);
+  let arguments = (ins AnyAttr:$shape,
+    StrAttr:$name, OptionalAttr<AnyAttr>:$value, OptionalAttr<I64Attr>:$offset);
```

Comment:
name == fileName? 
offset == byteOffsetInFile?

more expressive name will help clarity.
Agree that the name of the op should be changed to KrnlConstantOp